contracts/example/SnapshotGate/SnapshotGate.sol

```diff
@@ -0,0 +1,335 @@
+     *
+     * Reverts if:
+     * - Caller is not contract owner
+     * - Snapshot is already frozen
+     */
+    function freezeSnapshot() external onlyOwner {
+        // Make sure snapshot isn't frozen
+        require(!snapshotFrozen, "Snapshot already frozen");
+
+        // Freeze
+        snapshotFrozen = true;
+
+        // Notify watchers of state change
+        emit SnapshotFrozen();
+    }
+
+    /**
+     * @notice Commit to a gated offer on the Boson Protocol
+     *
+     * Commit to the specified offer on behalf of the buyer,
+     * first checking that the buyer is in the snapshot and
+     * hasn't already used all their available commits.
+     *
+     * Payment must be arranged it the token specified by the
+     * given offer.
+     *
+     * If price is set in the native token, e.g., MATIC on
+     * Polygon, it should be sent to this method in msg.value.
+     *
+     * For all other tokens, advance approval should be done
+     * to allow the protocol to transfer the caller's tokens to
+     * accept payment.
+     *
+     * Reverts if:
+     * - Snapshot is not frozen
+     * - Buyer doesn't have a balance of the given token in the snapshot
+     * - Buyer's balance of the given token in the snapshot has been used
+     * - insufficient payment or transfer not approved
+     * - offer is from another seller
+     * - The protocol reverts for any reason, including but not limited to:
+     *   - invalid offerId
+     *   - offer condition does not specify this contract as conditional token
+     *   - token id supplied to this method is not the id in the offer condition
+     *   - sold out - offer qty available did not match total supply of snapshot token
+     *
+     * @param _buyer the buyer address
+     * @param _offerId the id of the offer to commit to
+     * @param _tokenId the snapshot token the buyer is using for the commit
+     */
+    function commitToGatedOffer(
+        address payable _buyer,
+        uint256 _offerId,
+        uint256 _tokenId
+    ) external payable statusCheck {
+        // Make sure snapshot is frozen
+        require(snapshotFrozen, "Snapshot is not frozen");
+
+        // Find out how many tokens the buyer had at time of snapshot
+        uint256 owned = snapshot[_tokenId][_buyer];
+        require(owned > 0, "Buyer held no balance of the given token id at time of snapshot");
+
+        // Find out how many commits the buyer has done already
+        uint256 used = committed[_tokenId][_buyer];
+        require(owned > used, "Buyer's balance of the snapshot token id has been used");
+
+        // Store the details of the transaction
+        txDetails = TransactionDetails(_buyer, _tokenId);
+
+        // Track the usage
+        committed[_tokenId][_buyer] = ++used;
+
+        // Get the offer
+        bool exists;
+        Offer memory offer;
+        (exists, offer, , , , ) = IBosonOfferHandler(protocol).getOffer(_offerId);
```

Comment:
`exists` is not used anywhere. Either require that it's true or simply omit it.

Reply:
added a require